### PR TITLE
Handle dependencies with no exports

### LINKED ISSUE #1503: lintr::object_length_linter() does not work in the case of dependencies with no exports
Test package:

https://github.com/Bisaloo/testwpp

Error: 

> .Error in data.frame(pkg = ns, fun = getNamespaceExports(ns), stringsAsFactors = FALSE) : 
  arguments imply differing number of rows: 1, 0

This happens because the wpp2017 package does not export any functions. It only provides data. If I understand correctly, this means that the following line

https://github.com/r-lib/lintr/blob/f6cf291a9b28b8c0bf88928aa5167b0396ad687d/R/namespace.R#L29

will try to create a data.frame with one element in the first column and 0 in the second, hence the error.

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Closes #1503

### CODE REVIEW

File: R/namespace.R [hunk 1]
```diff
@@ -36,14 +41,18 @@ empty_namespace_data <- function() {
...
-    function(i) {
-      fun_obj <- get(ns_imports$fun[i], envir = asNamespace(ns_imports$pkg[i]))
-      is.function(fun_obj) && is_s3_generic(fun_obj)
-    },
-    logical(1L)
-  )
+  is_generic <- logical(0L)
+
+  if (!is.null(ns_imports)) {
+    is_generic <- vapply(
```
Comment (MichaelChirico) [inline R/namespace.R:47]: how about just using seq_len(NROW(ns_imports))?

Comment (MichaelChirico) [reply R/namespace.R:47]: alternatively, return a 0-row data.frame instead of NULL

Comment (IndrajeetPatil) [reply R/namespace.R:47]: That's cool! Thanks. 

I also added a comment; lemme know if it looks okay to you.

Comment (MichaelChirico) [reply R/namespace.R:47]: I think too verbose tbh

# NROW for the NULL case of 0-export dependencies, #1503

oughta do

**Review by MichaelChirico — COMMENTED** [review]
Now for the fun part -- could you please add a test?

FWIW it may be possible to use base datasets package

**Comment by IndrajeetPatil** [comment]
> FWIW it may be possible to use base datasets package

Sounds good. Will try to add a test.

**Comment by IndrajeetPatil** [comment]
@MichaelChirico I confirmed that the test fails on `main` branch.

File: tests/testthat/test-object_length_linter.R [hunk 2]
```diff
@@ -43,7 +43,22 @@ test_that("lints S3 generics correctly", {
 
 test_that("object_length_linter won't fail if an imported namespace is unavailable", {
   expect_length(
-    lint_package(test_path("dummy_packages", "missing_dep"), linters = object_length_linter(), parse_settings = FALSE),
+    lint_package(
+      test_path("dummy_packages", "missing_dep"),
+      linters = object_length_linter(),
```
Comment (MichaelChirico) [inline tests/testthat/test-object_length_linter.R:48]: the original issue is about object_usage_linter, right?

Comment (IndrajeetPatil) [reply tests/testthat/test-object_length_linter.R:48]: No, it's about `object_length_linter()`: https://github.com/r-lib/lintr/issues/1503

**Pull request merged**